From 77791b3f5b817becc9b1697c860b300fc8df1590 Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Mon, 20 Jul 2020 02:56:00 +0000
Subject: [PATCH 2/2] net: mscc: ocelot: drop the unused codes to fix build
 warnings

In ocelot driver, the function ocelot_ace_ptp_rule and variable
ptp_rule in function ocelot_init are not used at all. So drop them
to fix the compile warnings.

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/mscc/ocelot.c | 27 ---------------------------
 1 file changed, 27 deletions(-)

diff --git a/drivers/net/ethernet/mscc/ocelot.c b/drivers/net/ethernet/mscc/ocelot.c
index 1dd76ecdaf47..09b8806f4489 100644
--- a/drivers/net/ethernet/mscc/ocelot.c
+++ b/drivers/net/ethernet/mscc/ocelot.c
@@ -1763,35 +1763,8 @@ int ocelot_probe_port(struct ocelot *ocelot, u8 port,
 }
 EXPORT_SYMBOL(ocelot_probe_port);
 
-static struct ocelot_ace_rule *ocelot_ace_ptp_rule(struct ocelot *ocelot)
-{
-	struct ocelot_ace_rule *rule;
-
-	rule = kzalloc(sizeof(*rule), GFP_KERNEL);
-	if (!rule)
-		return rule;
-
-	/* Entry for PTP over Ethernet (etype 0x88f7)
-	 * Action: trap to CPU port
-	 */
-	rule->ocelot = ocelot;
-	rule->prio = 1;
-	rule->type = OCELOT_ACE_TYPE_ETYPE;
-	/* Available on all ingress port except CPU port */
-	rule->ingress_port = ~BIT(ocelot->num_phys_ports);
-	rule->dmac_mc= OCELOT_VCAP_BIT_1;
-	rule->frame.etype.etype.value[0] = 0x88;
-	rule->frame.etype.etype.value[1] = 0xf7;
-	rule->frame.etype.etype.mask[0] = 0xff;
-	rule->frame.etype.etype.mask[1] = 0xff;
-	rule->action = OCELOT_ACL_ACTION_TRAP;
-
-	return rule;
-}
-
 int ocelot_init(struct ocelot *ocelot)
 {
-	struct ocelot_ace_rule *ptp_rule;
 	u32 port;
 	int i, cpu = ocelot->num_phys_ports;
 	char queue_name[32];
-- 
2.17.1

